instrumentation/pg/lib/opentelemetry/instrumentation/pg/patches/connection.rb

```diff
@@ -86,7 +86,7 @@ def span_attrs(kind, *args)
             attrs = { 'db.operation' => validated_operation(operation), 'db.postgresql.prepared_statement_name' => statement_name }
             attrs['db.statement'] = sql unless config[:db_statement] == :omit
             attrs.merge!(OpenTelemetry::Instrumentation::PG.attributes)
-            attrs.reject! { |_, v| v.nil? }
+            attrs.compact!
```

Comment:
is there any perf benefit to inlining this and not using a destructive `!` method?, ie in the line below

```
            [span_name(operation), client_attributes.merge(attrs.compact)]
```

Reply:
I think instrumentations should limit the amount of Objects they create since it will contribute to triggering GC and require more memory.

Reply:
ok so then uh...perf regression. yea let's not do that lol

Reply:
Checkout cd98a8aad8bf044a0d106ab2d2ecdb7a7ffc38a9